core/src/main/java/apoc/export/graphml/XmlGraphMLReader.java

```diff
@@ -277,12 +277,13 @@ public long parseXML(Reader input) throws XMLStreamException {
     private RelationshipType getRelationshipType(XMLEventReader reader) throws XMLStreamException {
         if (this.labels) {
             XMLEvent peek = reader.peek();
-            if (peek.isCharacters() && !(peek.asCharacters().isWhiteSpace())) {
+            boolean isChar = peek.isCharacters();
+            if (isChar && !(peek.asCharacters().isWhiteSpace())) {
                 String value = peek.asCharacters().getData();
                 String el = ":";
                 String typeRel = value.contains(el) ? value.replace(el, StringUtils.EMPTY) : value;
                 return RelationshipType.withName(typeRel.trim());
-            } else {
+            } else if (!peek.isEndDocument() && !"data".equals(isChar ? peek.asCharacters().toString() : peek.asStartElement().getName().getLocalPart()) ) {
```

Comment:
che cos'è questo `peek.asStartElement().getName().getLocalPart()`?

Reply:
Quando arriva agli `edges`, nel caso ad esempio di `<data key="weight">1</data>` prende il nome del tag come stringa, cioè "data" in questo caso.